### PR TITLE
[closes #3805] Add platform HTTP security example

### PR DESCRIPTION
**Pull request by llowinge**
Closes https://github.com/apache/camel-quarkus/issues/3805.

### CODE REVIEW

File: docs/modules/ROOT/attachments/examples.json [hunk 1]
```diff
@@ -54,6 +54,11 @@
     "description": "Demonstrates how to add support for metrics, health checks and distributed tracing",
     "link": "https://github.com/apache/camel-quarkus-examples/tree/main/observability"
   },
+  {
+    "title": "Platform HTTP security",
+    "description": "Demonstrates how to secure platform HTTP with Kyecloak",
```
Comment (jamesnetherton) [inline docs/modules/ROOT/attachments/examples.json:59]: Maybe change the title to `Platform HTTP security with Keycloak`.

Also there's a typo in the description`Kyecloak`.

File: platform-http-security/README.adoc [hunk 2]
```diff
@@ -0,0 +1,272 @@
+= Platform HTTP security with Keycloak: A Camel Quarkus example
+:cq-example-description: An example that shows how to secure platform HTTP with Kyecloak
```
Comment (jamesnetherton) [inline platform-http-security/README.adoc:2]: Typo `Kyecloak`

File: platform-http-security/README.adoc [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+$ kubectl create secret generic quarkus-keycloak --from-literal=QUARKUS_OIDC_CREDENTIALS_SECRET=<YOUR_SECRET>
+$ mvn clean package -DskipTests -Dquarkus.kubernetes.env.vars.QUARKUS_OIDC_AUTH_SERVER_URL=$BASE_KEYCLOAK_KUBERNETES_URL/realms/quarkus -Dquarkus.oidc.tls.verification=none -Dquarkus.kubernetes.ingress.expose=true -Dquarkus.kubernetes.deploy=true -Dkubernetes
+----
+
+The `kubernetes` profile uses quarkus kubernetes and openshift-container extensions, as described in the `pom.xml`.
+
+[source,xml]
+----
+<dependencies>
+        <dependency>
```
Comment (jamesnetherton) [inline platform-http-security/README.adoc:168]: Nitpick - please can you fix the indentation of `<dependency>`.

File: platform-http-security/pom.xml [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-microprofile-health</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-platform-http</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-log</artifactId>
```
Comment (jamesnetherton) [inline platform-http-security/pom.xml:86]: I guess we don't need `log` or `direct` extensions? If not, the dependencies can be removed.

File: platform-http-security/README.adoc (hunk 2 above)
Comment (llowinge) [reply platform-http-security/README.adoc]: Thanks, resolved.

File: platform-http-security/pom.xml (hunk 4 above)
Comment (llowinge) [reply platform-http-security/pom.xml]: Thanks, resolved.

**Comment by jamesnetherton** [comment]
@llowinge I forgot there's a Maven mojo that syncs the examples JSON file. Run `mvn org.l2x6.cq:cq-maven-plugin:0.33.0:update-examples-json` then commit the changes.

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**